File: Beecrowd_Parte5/SurfAquatico_2560.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNext()) {
            int N = scanner.nextInt();
            int B = scanner.nextInt();
            int[] notas = new int[N];
            for (int i = 0; i < N; i++) {
                notas[i] = scanner.nextInt();
            }

            long resultado = 0;

            for (int i = 0; i <= N - B; i++) {
                int minNota = Integer.MAX_VALUE;
                int maxNota = Integer.MIN_VALUE;
                long soma = 0;

                for (int j = i; j < i + B; j++) {
                    soma += notas[j];
                    minNota = Math.min(minNota, notas[j]);
                    maxNota = Math.max(maxNota, notas[j]);
                }

                resultado += (soma - minNota - maxNota);
            }

            System.out.println(resultado);
        }
        scanner.close();
    }
}

File: Beecrowd_Parte5/SurfAquatico_2560.c
#include <stdio.h>
#include <limits.h>

#define MAXN 200005

int main() {
    int N, B;

    while (scanf("%d %d", &N, &B) != EOF) {
        int notas[MAXN];
        for (int i = 0; i < N; i++) {
            scanf("%d", &notas[i]);
        }

        long long resultado = 0;

        for (int i = 0; i <= N - B; i++) {
            int minNota = INT_MAX, maxNota = INT_MIN;
            long long soma = 0;

            for (int j = i; j < i + B; j++) {
                soma += notas[j];
                if (notas[j] < minNota) {
                    minNota = notas[j];
                }
                if (notas[j] > maxNota) {
                    maxNota = notas[j];
                }
            }

            resultado += (soma - minNota - maxNota);
        }

        printf("%lld\n", resultado);
    }

    return 0;
}
